### PR TITLE
Issue #5174 - Reorganized jetty-distribution

### LINKED ISSUE #5174: Reorganize jetty-distribution in version 10.0.0 to encourage jetty-home vs jetty-base split
**Jetty version**
10.0.0

**Description**

Proposed New Jetty Distribution.

```
/jetty-distribution-10.0.0/
    /LICENSE.md
    /README.md           - Explains the separation, and how to run the demo-base
    /jetty-home/
       /LICENSE.md
       /NOTICE.md
       /README.md
       /VERSION.txt
       /bin/
       /start.jar
       /etc/*
       /lib/*
       /modules/*
    /demo-base/
       /etc/*
       /resources/*
       /logs/*
       /webapps/*
```

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

**Comment by joakime** [comment]
I still need to update the README in the root of the jetty-distribution, just wanted to get something up here for others, so that we can start the conversation about this approach.

**Comment by janbartel** [comment]
FYI there's a couple of  problems in the reorganized distro:
*  the test-jndi webapp doesn't deploy. It can't find any of the resources that should be bound, and doesn't find javax.mail.Session class.
* there's demo-base/demo-base nested directory

**Comment by janbartel** [comment]
@joakime I pushed a fix to the correct locations for test-spec and test-jndi configs.

**Review by gregw — CHANGES_REQUESTED** [review]
as discussed:
```text

[152] git grep '\.\./start.jar' | cut -d: -f1 | uniq -c
      1 jetty-distribution/src/main/resources/README.TXT
      1 jetty-distribution/src/main/resources/start.ini
      1 jetty-documentation/src/main/asciidoc/distribution-guide/connectors/configuring-connectors.adoc
      1 jetty-documentation/src/main/asciidoc/distribution-guide/connectors/configuring-ssl-distribution.adoc
      1 jetty-documentation/src/main/asciidoc/distribution-guide/contexts/temporary-directories.adoc
      1 jetty-documentation/src/main/asciidoc/distribution-guide/logging/configuring-jetty-request-logs.adoc
     10 jetty-documentation/src/main/asciidoc/distribution-guide/logging/configuring-logging-modules.adoc
      2 jetty-documentation/src/main/asciidoc/distribution-guide/logging/default-logging-with-stderrlog.adoc
      2 jetty-documentation/src/main/asciidoc/distribution-guide/sessions/session-configuration-file-system.adoc
      3 jetty-documentation/src/main/asciidoc/distribution-guide/sessions/session-configuration-gcloud.adoc
      4 jetty-documentation/src/main/asciidoc/distribution-guide/sessions/session-configuration-hazelcast.adoc
      5 jetty-documentation/src/main/asciidoc/distribution-guide/sessions/session-configuration-infinispan.adoc
      2 jetty-documentation/src/main/asciidoc/distribution-guide/sessions/session-configuration-jdbc.adoc
      2 jetty-documentation/src/main/asciidoc/distribution-guide/sessions/session-configuration-mongodb.adoc
      3 jetty-documentation/src/main/asciidoc/distribution-guide/startup/custom-modules.adoc
      1 jetty-documentation/src/main/asciidoc/distribution-guide/startup/screen-list-modules.adoc
      1 jetty-documentation/src/main/asciidoc/distribution-guide/startup/start-jar.adoc
      5 jetty-documentation/src/main/asciidoc/distribution-guide/startup/startup-modules.adoc
      1 jetty-documentation/src/main/asciidoc/quickstart-guide/getting-started/jetty-running.adoc

```

File: jetty-distribution/src/main/resources/README.txt [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+and Servlet Container from the Eclipse Foundation
 
   http://www.eclipse.org/jetty/
 
-Jetty is open source and is dual licensed using the Apache 2.0 and
-Eclipse Public License 1.0.   You may choose either license when
-distributing Jetty.
+Eclipse Jetty is open source and is dual licensed using the
+Apache 2.0 and Eclipse Public License 2.0.
+You may choose either license when distributing Jetty.
```
Comment (gregw) [inline jetty-distribution/src/main/resources/README.txt]: At this point we need a section called "JETTY DISTRIBUTION" that describes what this directory is... how it contains a jetty-home (available separately), a demo-base and I think it should also contain a minimal jetty-base with just server module enabled.

We should make it clear that the jetty-home directory should be read-only and modifications should only be done to the base directories. Can we actually make jetty-home read-only in a portable way?

We could even say that to upgrade then only the jetty-home directory should be replaced with the jetty-home artefact.

File: jetty-distribution/src/main/resources/README.txt [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 
 
 RUNNING JETTY
```
Comment (gregw) [inline jetty-distribution/src/main/resources/README.txt:1]: I think the readme should then go as follows:
```
RUNNING JETTY
Assuming unix commands the general form of running jetty is:
   $ JETTY_HOME=/path/to/jetty-home
   $ cd /path/to/jetty-base
   $ java -jar $JETTY_HOME/start.jar

To see all the options to the start command:
  $java -jar $JETTY_HOME/start.jar --help

RUNNING THE DEMO_BASE
To see the configuration of the included demo-base
 $ cd demo-base
 $ java -jar $JETTY_HOME/start.jar --list-config

To run the demo (from the demo-base directory):
  $ java -jar $JETTY_HOME/start.jar

RUNNING A WAR
This distribution contains a jetty-base directory with a minimal configuration.
To enable http and webapp deployment for this base
  $ JETTY_HOME=$PWD/jetty-home
  $ cd jetty-base
  $ java -jar $JETTY_HOME/start.jar --add-to-start=http,deploy
  $ cp /path/to/mywebapp.war webapps

To see what other modules can be configured:
  $ java -jar $JETTY_HOME/start.jar --list-modules

This war in this base can then be run with
 $ java -jar $JETTY_HOME/start.jar

CREATING A NEW JETTY BASE
A new Jetty base can be created anywhere on the file system and with any name:
 $ mkdir /path/to/my-jetty-base
 $ cd /path/to/my-jetty-base
 $ java -jar $JETTY_HOME/start.jar --create-startd --add-to-start=server,http,deploy
 $ cp /path/to/mywebapp.war webapps

This base can then be run with

 $ java -jar $JETTY_HOME/start.jar

```

I don't think we really need the rest of the readme other than links to the documentation!

**Review by gregw — CHANGES_REQUESTED** [review]
I like this a lot, but a few more changes... mostly to the README.
I also think we need a README in the demo-base directory that describes exactly what it demos and how it was setup.

**Review by olamy — APPROVED** [review]
👍

File: jetty-distribution/src/main/resources/README.txt (hunk 2 above)
Comment (joakime) [reply jetty-distribution/src/main/resources/README.txt:1]: Done.
I'm good with what you wrote (just some minor formatting niggles)

**Pull request merged**